ca/server.go

```diff
@@ -22,6 +22,50 @@ const (
 	defaultReconciliationRetryInterval = 10 * time.Second
 )
 
+// UpdateSecurityConfigFromCluster takes an api.Cluster object and updates the security config based on the changes
+// in the RootCA and ExternalURLs from that cluster object.
+func UpdateSecurityConfigFromCluster(ctx context.Context, s *SecurityConfig, cluster *api.Cluster) error {
```

Comment:
Maybe just `UpdateSecurityConfig`?

Would it make sense to make it a method on `(*SecurityConfig)`? and call it `Update`?